nx-X11/programs/Xserver/hw/nxagent/Window.c

```diff
@@ -3111,6 +3135,18 @@ FIXME: Do we need to set save unders attribute here?
     }
   }
 
+  if ((nxagentRealWindowProp) && (!nxagentWindowTopLevel(pWin)))
+  {
+    Atom prop = MakeAtom("NX_REAL_WINDOW", strlen("NX_REAL_WINDOW"), True);
+
+    if (ChangeWindowProperty(pWin, prop, XA_WINDOW, 32, PropModeReplace, 1, nxagentWindowPriv(pWin), 1) != Success)
+        fprintf(stderr, "nxagentReconnectWindow: Updating NX_REAL_WINDOW failed.\n");
+#ifdef DEBUG
```

Comment:
>> If the update fails the property should better be deleted to avoid confusion.

> Indeed. Seeing into that.

Not needed. On window creation, the code tries to add NX_REAL_WINDOW. If that fails, the property will be missing.

Once we suspend, the property gets removed.

If we reconnect the property is removed then. If we fail to re-add it, then it is still not present. Voilà!